src/app/services/shared/bucket_operations_service.py

```diff
@@ -0,0 +1,17 @@
+import json
+
+from google.cloud.storage.bucket import Bucket
+
+
+class BucketOperationsService:
+    @staticmethod
+    def get_bucket_file_as_json(filename: str, bucket: Bucket) -> object:
+        """
+        Gets a file from a google cloud bucket with a specific filename and loads it as json.
+
+        Parameters:
+        filename (str): name of file being loaded.
+        bucket (Bucket): bucket being accessed.
+        """
+
+        return json.loads(bucket.blob(filename).download_as_string())
```

Comment:
I am thinking that it will be better to retain the try block here to intercept the exception of not able to find the associated file from bucket, or else it will trigger a 500 error instead of 404.
Although it should be abnormal case to not able to find the file when metadata record exists in FireStore, but returning only a 500 error with no specific logs may hinder the debug process when it happens

Reply:
Good point. I've added it to SchemaBucketRepository instead of here though, since this function is used by both the schema bucket and dataset bucket. If I added it here I'd need to create a new exception handler for a generic bucket query exception which I think is beyond the scope of this (already bloated) ticket.